docker-compose.yml

```diff
@@ -1,133 +1,145 @@
-  #     - "cluster.name=es-mastodon"
-  #     - "discovery.type=single-node"
-  #     - "thread_pool.write.queue_size=1000"
-  #   networks:
-  #      - external_network
-  #      - internal_network
-  #   healthcheck:
-  #      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
-  #   volumes:
-  #      - ./elasticsearch:/usr/share/elasticsearch/data
-  #   ulimits:
-  #     memlock:
-  #       soft: -1
-  #       hard: -1
-  #     nofile:
-  #       soft: 65536
-  #       hard: 65536
-  #   ports:
-  #     - '127.0.0.1:9200:9200'
+x-image-tag: &image-tag nightly
 
+services:
   web:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.11
-    restart: always
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:*image-tag
     env_file: .env.production
+    environment:
+      - PORT=3000
+      - BIND=127.0.0.1
+    restart: always
     command: bundle exec puma -C config/puma.rb
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          "curl -s --noproxy localhost localhost:3000/health | grep -q 'OK' || exit 1",
+        ]
     networks:
       - external_network
       - internal_network
-    healthcheck:
-      # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:3000/health || exit 1']
     ports:
       - '127.0.0.1:3000:3000'
+    volumes:
+      - ./public/system:/mastodon/public/system
     depends_on:
       - db
       - redis
-      # - es
+      - es
+
+  sidekiq:
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon:*image-tag
+    env_file: .env.production
+    restart: always
+    command: bundle exec sidekiq
+    healthcheck:
+      test: ['CMD-SHELL', "ps aux | grep '[s]idekiq\ 6' || false"]
+    networks:
+      - external_network
+      - internal_network
     volumes:
       - ./public/system:/mastodon/public/system
+    depends_on:
+      - db
+      - redis
+      - es
 
   streaming:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.11
-    restart: always
+    # build:
+    #   context: ./streaming
+    #   dockerfile: Dockerfile
+    image: ghcr.io/mastodon/mastodon-streaming:*image-tag
     env_file: .env.production
-    command: node ./streaming
+    environment:
+      - PORT=4000
+      - BIND=127.0.0.1
+    restart: always
+    command: node ./streaming/index.js
+    healthcheck:
+      test:
+        [
+          'CMD-SHELL',
+          "curl -s --noproxy localhost localhost:4000/api/v1/streaming/health | grep -q 'OK' || exit 1",
+        ]
     networks:
       - external_network
       - internal_network
-    healthcheck:
-      # prettier-ignore
-      test: ['CMD-SHELL', 'wget -q --spider --proxy=off localhost:4000/api/v1/streaming/health || exit 1']
     ports:
       - '127.0.0.1:4000:4000'
     depends_on:
       - db
       - redis
 
-  sidekiq:
-    build: .
-    image: ghcr.io/mastodon/mastodon:v4.2.11
+  db:
```

Comment:
I am not a big fan of having those services in docker, because it makes it harder for people to update them. Maybe we should keep those here in the long run and provide instructions on how to upgrade major pg versions, but I feel more confident in having pg as a system service (or hosted service), and let those packages handle the major version upgrades

Reply:
I'm all for removing them, and having directions on where to find information on how to do that separately or incorporate them into the file on their own.